### PR TITLE
Renamed metrics::Recorder & metrics-runtime::Sink methods

### PR DESCRIPTION
**Pull request by jean-airoldie**
* Renamed record_counter to increment_counter
* Renamed record_gauge to update_gauge

### CODE REVIEW

File: metrics/src/lib.rs [hunk 1]
```diff
@@ -324,12 +324,12 @@ pub fn recorder() -> &'static dyn Recorder {
 
 #[doc(hidden)]
 pub fn __private_api_record_count(key: Key, value: u64) {
```
Comment (tobz) [inline metrics/src/lib.rs]: `__private_api_record_count` -> `__private_api_increment_counter`?

Comment (jean-airoldie) [reply metrics/src/lib.rs]: missed that

**Comment by tobz** [comment]
Should we also update the documentation for `gauge!`, `timing!`, and `value!`? The stuff you did in your last PR: changing the wording, fixing the macro import, etc.

**Comment by jean-airoldie** [comment]
This breaks however the links in `metrics` doc, so Ill fix that.

**Pull request merged**